src/ol/source/xyzsource.js

```diff
@@ -63,6 +69,16 @@ goog.inherits(ol.source.XYZ, ol.source.TileImage);
 
 
 /**
+ * Return the URLs used for this XYZ source.
+ * @return {!Array.<string>|null} URLs.
+ * @api
+ */
+ol.source.XYZ.prototype.getUrls = function() {
+  return goog.isDefAndNotNull(this.urls_) ? this.urls_ : null;
```

Comment:
You can just return `this.urls_` here - it will be null already if it was not configured.